mindsdb/integrations/handlers/ollama_handler/ollama_handler.py

```diff
@@ -0,0 +1,127 @@
+import json
+import requests
+from typing import Dict, Optional
+
+import pandas as pd
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.integrations.libs.llm_utils import get_completed_prompts
+
+
+class OllamaHandler(BaseMLEngine):
+    name = "ollama"
+    SERVE_URL = 'http://localhost:11434'
+    MODEL_LIST_URL = 'https://registry.ollama.ai/v2/_catalog'
+
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+        if 'using' not in args:
+            raise Exception("Ollama engine requires a USING clause! Refer to its documentation for more details.")
+        else:
+            args = args['using']
+
+        # check model version is valid
+        all_models = requests.get(OllamaHandler.MODEL_LIST_URL).json()['repositories']
```

Comment:
Maybe wrap this in try/except in case of connectivity issues

Reply:
Good point!